org.adempiere.base/src/org/adempiere/util/ModelClassGenerator.java

```diff
@@ -168,6 +168,7 @@ private String createHeader (int AD_Table_ID, StringBuilder sb, StringBuilder ma
 		start.append("/** Generated Model for ").append(tableName).append(NL)
 			 .append(" *  @author iDempiere (generated) ").append(NL)
 			 .append(" *  @version ").append(Adempiere.MAIN_VERSION).append(" - $Id$ */").append(NL)
+			 .append("@org.adempiere.base.Model(table=\"").append(tableName).append("\", intermediate=true)").append(NL)
 			 .append("public class ").append(className)
 			 	.append(" extends PO")
 			 	.append(" implements I_").append(tableName)

```

Comment:
I think we need to have the facility to generate the M class here (or add annotation to existing M class). Without this, we are leave with generated class that can't be used by DefaultModelFactory or accidentally take precedence over existing M* class (if we take the intermediate class).